go/consensus/cometbft/full/archive.go

```diff
@@ -236,3 +246,34 @@ func NewArchive(
 
 	return srv, srv.initialize()
 }
+
+// serviceClientWorker handles command dispatching.
+func (t *archiveService) serviceClientWorker(ctx context.Context, svc api.ServiceClient) {
+	defer t.serviceClientsWg.Done()
+
+	sd := svc.ServiceDescriptor()
+	if sd == nil {
+		// Some services don't actually need a worker.
+		return
+	}
+
+	logger := t.Logger.With("service", sd.Name())
+	logger.Info("starting command dispatcher")
+
+	// Service client event loop.
+	var height int64
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case cmd := <-sd.Commands():
```

Comment:
The `serviceClientWorker` in the non-archive full node first waits for a block before processing commands. Not sure if this is relevant here.

Reply:
Probably since a non-archive node can be without blocks, and it needs the height of the latest block to be passed in the `DeliverCommand`. I have updated the archive node worker to also query the latest block on start and use the height in the `DeliverCommand`.